File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.js
// Last updated: 3/6/2026, 2:08:53 PM
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
    let greatestCandies = 0;
    for (let i = 0; i < candies.length; i++) {
        if (candies[i] > greatestCandies) {
            greatestCandies = candies[i];
        }
    }
    let result = [];

    for(let i=0;i<candies.length;i++){
        if(candies[i] + extraCandies >= greatestCandies){
            result.push(true);
        }
        else{
            result.push(false);
        }
    }
    return result;
};

File: 1528-KidsWithTheGreatestNumberOfCandies/1528-KidsWithTheGreatestNumberOfCandies.cpp
// Last updated: 7/17/2026, 4:26:21 PM
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max = candies[0];

        for(int i=1;i<candies.size();i++){
            if(candies[i]>max){
                max=candies[i];
            }
        }

        vector<bool> result;

        for(int i=0;i<candies.size();i++){
            if(candies[i]+extraCandies >= max){
                result.push_back(true);
            }
            else{
                result.push_back(false);
            }
        }
        return result;
    }
};